google/cloud/pubsub_v1/subscriber/message.py

```diff
@@ -242,9 +243,44 @@ def ack(self) -> None:
                 byte_size=self.size,
                 time_to_ack=time_to_ack,
                 ordering_key=self.ordering_key,
+                future=None
             )
         )
 
+    def ack_with_response(self) -> None:
+        """Acknowledge the given message.
```

Comment:
The only difference compared to `ack()` is whether or not we use and return a future. It would be IMO better for maintenance if we instead just add an optional `with_response` argument to `ack()` (`False` by default) and set the value of `future` accordingly.

Reply:
I agree with 3 or more, while two instances of the same are sometimes borderline. I would personally still go with de-duplication here, but I also understand a different view.